packages/console/src/pages/AuthStatusChecker/index.tsx

```diff
@@ -0,0 +1,392 @@
+  const [debugLogs, setDebugLogs] = useState<DebugLogEntry[]>([]);
+  const [isDebugMode, setIsDebugMode] = useState(false);
+  const [currentToken, setCurrentToken] = useState<string | undefined>(undefined);
+
+  const urlParams = new URLSearchParams(window.location.search);
+  const debugParam = urlParams.get('debug');
+  const debugFlag = localStorage.getItem(DEBUG_MODE_KEY);
+
+  // Check if we're in debug mode (query parameter or localStorage flag)
+  useEffect(() => {
+    if (yes(debugParam) || yes(debugFlag)) {
+      setIsDebugMode(true);
+    }
+  }, [debugParam, debugFlag]);
+
+  // Add debug log entry
+  const addDebugLog = useCallback(
+    (type: DebugLogEntry['type'], message: string, data?: unknown) => {
+      // Skip logging if debug mode is not enabled.
+      if (!isDebugMode) {
+        return;
+      }
+
+      const logEntry: DebugLogEntry = {
+        timestamp: new Date().toISOString(),
+        type,
+        message,
+        data,
+      };
+
+      console.log(`[AuthStatusChecker] ${type.toUpperCase()}: ${message}`, data);
+      setDebugLogs((previous) => [...previous, logEntry]);
+    },
+    [isDebugMode]
+  );
+
+  // Monitor token changes
+  useEffect(() => {
+    const token = localStorage.getItem(ADMIN_CONSOLE_ID_TOKEN_KEY);
+    setCurrentToken(token ?? undefined);
+    addDebugLog('info', `Current token status: ${token ? 'present' : 'absent'}`, {
+      tokenLength: token?.length,
+    });
+  }, []);
+
+  useEffect(() => {
+    addDebugLog('info', 'AuthStatusChecker component mounted', {
+      origin: window.location.origin,
+      hostname: window.location.hostname,
+      isInIframe: window !== window.top,
+      userAgent: navigator.userAgent,
+    });
+
+    const handleMessage = async (event: MessageEvent) => {
+      addDebugLog('received', `Message from ${event.origin}`, {
+        origin: event.origin,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        data: event.data,
+        source: event.source === window.parent ? 'parent' : 'other',
+      });
+
+      // For security, we should validate the origin
+      const currentOrigin = window.location.origin;
+      const currentHostname = window.location.hostname;
+      const eventOriginUrl = new URL(event.origin);
+      const eventHostname = eventOriginUrl.hostname;
+
+      // Allow communication if:
+      // 1. Same origin (exact match)
+      // 2. Same top-level domain (e.g., *.domain.com)
+      // 3. Both are localhost (for development)
+      const isValidOrigin =
+        event.origin === currentOrigin || // Same origin
+        (currentHostname === 'localhost' && eventHostname === 'localhost') || // Both localhost
+        // Same top-level domain check
+        (currentHostname.includes('.') &&
+          eventHostname.includes('.') &&
+          currentHostname.split('.').slice(-2).join('.') ===
+            eventHostname.split('.').slice(-2).join('.'));
```

Comment:
nit: this domain check doesn't cover multi-level TLDs like `.co.uk`, which is fine for our current scope but good to be aware of for the future.

Reply:
Updated, using `pls`